### DISCUSSION TITLE
Issue Template forms:  Allow validation rules (maxLength, regex etc)

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [community/community#10227] [Projects and Issues]
I absolutely love Github's new Issue template forms ❤️ 
They have helped increase the quality of issue reports significantly. For example, if you look at the following issues in these repos from the last month (_since we added Template forms_), user issued reports have increased significantly, and even as a non-maintainer of some repos, I can follow along & learn from these reproducible code examples members of the community are sharing in their issue reports:
- https://github.com/antvis/G6/issues
- https://github.com/tannerlinsley/react-query/issues

## Problem Statement
**Currently there is no way to add more granular validation rules to `ISSUE_TEMPLATE` file.**
As a result of this, people can dupe required fields in the template form. For example in the image below, a user submitted a codesanbox link which is not valid.

 
<img src="https://user-images.githubusercontent.com/6743796/149785431-ade73db4-6ad3-413e-afe6-74bc2eb7683c.png" width="500"/>

### This is a big problem. How/Why? 
To give some context for readers who stumble upon this post I'm writing:

In frontend/UI related repositories, we require user's to submit a `codesanbox.io` URL to a small reproducible code example. The purpose of this is many:
 - **help issue reporters create a very small reproducible code example** of their code that is **easily runnable & able to be looked within < 1 minute** by a project owner or other member of the community.
 - it incentivizes the user to REALLY think about the issue & break down their challenge. 
     - If they are able to do this, it is very likely that the bug report will be high quality & that the issue they have is really important to them. 
     - it is also increases the chance that a member of the community is able to look at the code & help **because they've made the process easy**
 - helps **significantly filter out issues with little or no context, details and reproducible information**
     - understandably, when you hit a challenge with some code & are frustrated its too easy to go to a Github repo & submit a bug report with little context under frustration. We understand that dedicating the time to create a detailed bug report or reproducible code example takes time. 
          -  _I've often spent between 30mins - 1 hour creating a reproducible code example & before starting thinking "...ahh I feel pressed for time today...forget writing up a detailed bug report...", but I've gotten use to the habit of following through fortunately._ 
          - **Repo owners/project maintainers which are the backbone of Github**, shouldn't suffer the burden of losing time & mental bandwidth of engaging in preventable back & forth conversations asking for more detailed information from user's.
 
 ## Solutions Considered
1.  Allow the users to specify a regex rule in the `ISSUE_TEMPLATE` file:

 
```yml
   validations:
      required: true
      validationRule: /some_regex_here/   <----- ✅ 🙏
```

2. Provide an an easy API to create custom ISSUE_TEMPLATE forms ?
- The [Ant Design](https://github.com/ant-design/ant-design/) team created their own Github Issue template form to add more validation's & their own custom logic.

https://user-images.githubusercontent.com/6743796/149796397-af08fd13-dad4-427d-93a5-de19faae0bf1.mp4

 
 
## Example of Issue in Real life:

<img src="https://user-images.githubusercontent.com/6743796/149785431-ade73db4-6ad3-413e-afe6-74bc2eb7683c.png" width="500"/>

<img width="585" alt="CleanShot 2022-01-17 at 06 14 40@2x" src="https://user-images.githubusercontent.com/6743796/149785758-5768217f-d4e5-49eb-8163-bb61ffdef569.png">

## Related Discussions
- @pglevy https://github.com/github/feedback/discussions/10101
- https://twitter.com/TkDodo/status/1482956284429316097

## Additional Resources
In case anyone from the community wants to learn more about Github issue templates check out the links below 😄 
- [official github docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository)
- a real example of an awesome github `ISSUE_TEMPLATE` for copy-pasting ([ISSUE_TEMPLATE/bug_report.yml](https://github.com/tannerlinsley/react-query/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml) )

   <details>
      <img src="https://user-images.githubusercontent.com/6743796/149788081-1081b581-d94c-4764-9cb4-3d3a9aa95a7d.png" />
    </details>

**Comment by cliffordfajardo** [comment]
I've mentioned a few people below who are very active & maintain some of the most widely popular github repositories in their respective areas. They will probably have a better sense of what would be helpful & may have better ideas than what I mentioned 😅 :

[react-query](https://github.com/tannerlinsley/react-query): @tkdodo , @tannerlinsley, @arnaudbzn
[react-flow](https://github.com/wbkd/react-flow): @moklick @chrtze
[react](https://github.com/facebook/react) - @bvaughn, @gaearon
[ant-design](https://github.com/ant-design/ant-design/): @afc163, @zombieJ
[vue](https://github.com/vuejs/vue): @yyx990803
[fastify](https://github.com/fastify/fastify): @mcollina

@anipind, @oneill38

**Reply by AtulGEM** [reply]
Hi Everyone , i was working on this and wanted to check if there is any progress done / any new update

**Comment by Ana06** [comment]
This would be very useful in https://github.com/mandiant/VM-Packages/issues/new?template=new_package.yml as we use GH actions to parse the issues and require for example that the package name does not have spaces. This is used to add new tools to the [flare-vm](https://github.com/mandiant/flare-vm) and [commando-vm](https://github.com/mandiant/commando-vm) projects.

In addition, I would also like to be able to validate the format of the issue title.
